geth/node/node.go

```diff
@@ -79,15 +79,15 @@ func MakeNode(config *params.NodeConfig) (*node.Node, error) {
 		if err := activateLightEthService(stack, config); err != nil {
 			return nil, fmt.Errorf("%v: %v", ErrLightEthRegistrationFailure, err)
 		}
-	} else {
-		// `personal_sign` and `personal_ecRecover` methods are important to
-		// keep DApps working.
-		// Usually, they are provided by an ETH or a LES service, but when using
-		// upstream, we don't start any of these, so we need to start our own
-		// implementation.
-		if err := activatePersonalService(stack, config); err != nil {
-			return nil, fmt.Errorf("%v: %v", ErrPersonalServiceRegistrationFailure, err)
-		}
+	}
+
+	// `personal_sign` and `personal_ecRecover` methods are important to
+	// keep DApps working.
+	// Usually, they are provided by an ETH or a LES service.
+	// Unfortunately, the parameters of `personal_sign` used in DApps
+	// are different from what LES provides, so we start this service always.
+	if err := activatePersonalService(stack, config); err != nil {
```

Comment:
If we activate Personal Service always now, I wonder if we can remove handlers and define these `personal_*` methods only in the service.

Reply:
`personal` namespace isn't exported via `CallRPC`, and it shouldn't.
Custom handlers right now serve the exact case to export 2 methods to `CallRPC` that is accessible to DApps.

Anyway, it looks like CK, for example, behaves very inconsistently with parameter flips, so this PR is blocked until we hear from their devs...

Reply:
Ok, I remember now. We have `sign` and `recover` exposed through handlers and these handlers call original `Personal` service API.